### PR TITLE
Build Boost.NumPy as different packages per different Python major versions

### PR DESCRIPTION
**Pull request by jslee02**
This is necessary to install two versions of Boost.NumPy for Python 2 and 3 side by side. Noost.NumPy dependent packages now should find them as
```cmake
find_package(BoostPython2Numpy) # for Python 2
find_package(BoostPython3Numpy) # for Python 3
```

This is a stopgap solution until we use Boost 1.63; Boost.Numpy will be shipped with Boost.Python from 1.63.

### CODE REVIEW

**Comment by jslee02** [comment]
[`ufunc.py` test is failing](https://travis-ci.org/personalrobotics/Boost.NumPy/jobs/299923990#L886), which I believe it is irrelevant to this change. Probably it's due to newer version of [`PythonInterp` (2.7.13)](https://travis-ci.org/personalrobotics/Boost.NumPy/jobs/299923990#L678) and [`NumPy` (1.13.1)](https://travis-ci.org/personalrobotics/Boost.NumPy/jobs/299923990#L681). The same test was passing with older versions of [`PythonInterp` (2.7.6)](https://travis-ci.org/personalrobotics/Boost.NumPy/jobs/295630914#L677) and [`NumPy` (1.8.2)](https://travis-ci.org/personalrobotics/Boost.NumPy/jobs/295630914#L679).

File: .travis.yml [hunk 1]
```diff
@@ -1,13 +1,12 @@
 sudo: required
 dist: trusty
-language: cpp
+language: python
```
Comment (psigen) [inline .travis.yml:3]: I don't see a reason to change this, we are doing a C++ library build after all.

File: .travis.yml [hunk 2]
```diff
@@ -1,13 +1,12 @@
...
 
 python:
   - "2.7"
   - "3.4"
 
 before_install:
-  - export PATH=$(echo $PATH | tr ':' "\n" | sed '/\/opt\/python/d' | tr "\n" ":" | sed "s|::|:|g")
```
Comment (psigen) [inline .travis.yml:10]: Was this just unused?

**Review by psigen — COMMENTED** [review]
Will this change break compatibility with anywhere else we might to use this library?  We are forking the original project in a way will probably never be backportable.

What (if any) are the ramifications if someone has another `Boost.Numpy` with the original naming present on their system?

**Comment by jslee02** [comment]
> Will this change break compatibility with anywhere else we might to use this library? We are forking the original project in a way will probably never be backportable.

That's true. But I don't believe we want to merge this into the upstream because the original project is deprecated and it will instead be shipped with Boost.Python since Boost 1.63. Once Boost 1.63 available, we might link to Boost.Python (including the current Boost.NumPy) something like:
```cmake
find_package(Boost COMPONENT python) # for Python 2
find_package(Boost COMPONENT python-py${PYTHON_MAJOR}${PYTHON_MINOR}) # for Python 3
```

> What (if any) are the ramifications if someone has another Boost.Numpy with the original naming present on their system?

It should be fine as long as they are installed manually instead of using PPA or other package managers. I don't expect (or haven't tested :sweat:) any conflict with another Boost.Numpy with the original naming because this change uses different package names.

If we try to install those packages using a package manager (at least `apt-get`), it might be a problem. Because they will install the same headers (binaries are different) and `apt-get` doesn't allow that try. However, I couldn't find any existing PPA that contains `Boost.Numpy` debian packages.

File: .travis.yml (hunk 1 above)
Comment (jslee02) [reply .travis.yml:3]: Sorry, I'm being lazy so didn't get the exact reason, but without this changing Travis CI doesn't start matrix build for
```yaml
python:
  - "2.7"
  - "3.4"
```

File: .travis.yml (hunk 2 above)
Comment (jslee02) [reply .travis.yml:10]: Frankly, I just didn't understand why we need this, but it seems to work without this line. I'm open to leaving this line as it is though.

File: .travis.yml (hunk 1 above)
Comment (psigen) [reply .travis.yml:3]: Ah, ok, that makes sense.

File: .travis.yml (hunk 2 above)
Comment (psigen) [reply .travis.yml:10]: Ok, just checking.  We should remove it, it looks like it's trying to do some weird path resolution to an `/opt/python` binary which we don't have.

**Comment by psigen** [comment]
> If we try to install those packages using a package manager (at least apt-get), it might be a problem. Because they will install the same headers (binaries are different) and apt-get doesn't allow that try. 

I was wondering, are we going to need to rename these header includes if _we_ are packaging this for PPA installation with dartpy?

Because if someone installs a `dartpy` for Python 2 and a `dartpy` for Python 3, we would end up with the same conflict--  we would need to install both Boost.Python versions, and then the headers would be installed to the same location, right?

Or is this somehow obviated because we would package a binary-only `Boost.Numpy` package and separate out the `include`s into a `Boost.Numpy-dev` package that would depend on either the Python 2 or Python 3 versions?

**Comment by jslee02** [comment]
Header files should be installed once. I'm thinking of the method you mentioned: Splitting the package into header package and binary packages, and making the two binary packages depends on the header package.

**Comment by jslee02** [comment]
@psigen Can we merge this?

**Comment by psigen** [comment]
@jslee02: Sorry, just saw this.

I think if we follow up with packaging the header split as you detailed in https://github.com/personalrobotics/Boost.NumPy/pull/4#issuecomment-344974466, then this should be fine.

**Review by psigen — APPROVED** [review]

**Pull request merged**